ml/src/test/scala/ai/h2o/sparkling/ml/algos/AnomalyPredictionTestSuite.scala

```diff
@@ -42,7 +45,12 @@ class AnomalyPredictionTestSuite
   override protected def expectedDetailedPredictionCol: StructField = {
     val scoreField = StructField("score", DoubleType, nullable = false)
     val normalizedScoreField = StructField("normalizedScore", DoubleType, nullable = false)
-    StructField("detailed_prediction", StructType(scoreField :: normalizedScoreField :: Nil), nullable = true)
+    val leafNodeAssignmentField =
+      StructField("leafNodeAssignments", ArrayType(StringType, containsNull = true), nullable = true)
```

Comment:
nit: Shouldn't  be `containsNull = false` ? (`ArrayType(StringType, containsNull = true)`)

Reply:
Yup, actually we are sure that there won't be any nulls in the array, but there is an (I would call it a bug) in Spark that when you define case class like `case class Detailed(assignments: Array[String])` and this udf is used to infer schema of produced dataframe, it always produces `containsNull = true`. 

So instead of trying to do some hacks around it, I decided to keep it simple and just define the nullability globally.

WDYT?

Reply:
Yup, that's fine. Spark and nullability is a big topic ...